compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -2273,6 +2274,107 @@ OMR::ResolvedMethodSymbol::setNoTemps(bool b)
+ * index which corresponds to the start of the basic block in which bytecodeIndex
+ * was generated at ilgen time. This is useful when trying to reassociate profiling
+ * information when blocks have been split due to inlining
+ *
+ *  @param bytecodeIndex The bytecode of the instruction whose basic block start is to be found
+ *  @return The bytecode index of the start of the containing ilgen basic block, -1 if unknown
+ */
+int32_t OMR::ResolvedMethodSymbol::getProfilingByteCodeIndex(int32_t bytecodeIndex)
+   {
+   for (auto itr = _bytecodeProfilingOffsets.begin(), end = _bytecodeProfilingOffsets.end(); itr != end; ++itr)
+      {
+      if (itr->first >= bytecodeIndex)
+         return itr->second.first;
+      }
+   return -1;
+   }
+
+/**
+ * \brief Save the profiling frequency of a given bytecode index
+ *
+ * This function stashes a profiling frequency against a given bytecode index for later retrieval
+ * (usually as part of profiling re-association.
+ *
+ * @param bytecodeIndex The bytecode index of the start of the block whose frequency we wish to save
+ * @param frequency The normalized block frequency to save for later use
+ */
+void OMR::ResolvedMethodSymbol::setProfilerFrequency(int32_t bytecodeIndex, int32_t frequency)
+   {
+   for (auto itr = _bytecodeProfilingOffsets.begin(), end = _bytecodeProfilingOffsets.end(); itr != end; ++itr)
+      {
+      if (itr->first >= bytecodeIndex)
+         itr->second.second = frequency;
+      }
+   }
+
+/**
+ * \brief Return the saved profilign frequency for the given bytecode index
+ *
+ * This function returns a profiling frequency that was saved against a given byte code index for
+ * later retrieval.
+ *
+ * @param bytecodeIndex The bytecode index of the start of the block whose frequency we wish to retrieve
+ * @return The normalized saved frequency, -1 if unknown
+ */
+int32_t OMR::ResolvedMethodSymbol::getProfilerFrequency(int32_t bytecodeIndex)
+   {
+   for (auto itr = _bytecodeProfilingOffsets.begin(), end = _bytecodeProfilingOffsets.end(); itr != end; ++itr)
+      {
+      if (itr->first >= bytecodeIndex)
+         return itr->second.second;
+      }
+   return -1;
+   }
+
+static bool compareProfilingOffsetInfo(const std::pair<int32_t, int32_t> &first, const std::pair<int32_t, int32_t> &second)
+   {
+   return first.first < second.first;
+   }
+
+/**
+ * \brief Save the bytecode range of a basic block during ilgen
+ *
+ * @param startBCI The bytecode index of the start of the basic block
+ * @param endBCI The bytecode index of the end of the basic block
+ */
+void OMR::ResolvedMethodSymbol::addProfilingOffsetInfo(int32_t startBCI, int32_t endBCI)
+   {
+   for (auto itr = _bytecodeProfilingOffsets.begin(), end = _bytecodeProfilingOffsets.end(); itr != end; ++itr)
+       {
+       if (itr->first >= endBCI)
+          {
+          _bytecodeProfilingOffsets.insert(itr, std::make_pair(endBCI, std::make_pair(startBCI, -1)));
```

Comment:
Could you please clarify what the -1 value for second.second represents here ? It seems to be set in setProfilerFrequency but because the usage of the api is not included in this commit (perfectly fine that it is'nt) I cannot see who sets that field value under what circumstance.

Possible to state up front what first, second.first and second.second denote at a high level in the summary for the PR ?